docs/user_guide/bulk_api_notes.rst

```diff
@@ -0,0 +1,38 @@
+
+Another, not so common cause, is to think to have the correct schema in place, while in fact you don't. For example, if you are trying to insert a record with a field that is a lookup to another object, but you are sending the wrong ID (external key or not), you will get this error. **Check that your schema is correct.** You might have two objects with the same name but different API names in Salesforce (for example: "Environment" from the DevOps Center managed package by salesforce and another "Environment" custom object that you have in your org, and when creating the lookup field in your other custom object, you accidentaly selected the managed package object instead of your custom object).
+
+Working with an output from Pandas
+==================================
+
+If you are working with the Pandas library in Python, you might want to use the `to_dict` method to convert your DataFrame to a dictionary. This is a good idea, but you need to be careful with the data types.
+
+For example, if you have a column that is going to be mapped to a Salesforce date/time field, you might want to convert it to the ISO format before uploading the data with the Bulk API. There are many ways of converting a number or string to the ISO format, but here's an example:
+
+.. code-block:: python
+
+    my_dataframe['CreatedDate'] = my_dataframe['CreatedDate'].dt.strftime('%Y-%m-%dT%H:%M:%S.000Z')
+
+This will convert the `CreatedDate` column to the ISO format, which is one of the formats that Salesforce expects.
+
+Preparing the dataframe for the Bulk API if you are using external ids in a lookup relationship
+-----------------------------------------------------------------------------------------------
+
+If you are upserting data that looks up to existing data in Salesforce through an external id, make sure you define the dictionary object in the Python code before calling the *sf.bulk.* methods. For example, if you have a custom object called "MyCustomObject__c" that has a lookup relationship to the Account object, and you are using the `AccountNumber` field as the external id, you will need to define the dictionary object like this:
+
+.. code-block:: python
+
+    my_dataframe['Account__r'] = my_dataframe['Account__r'].apply(lambda x: {'AccountNumber': x})
+
+This will convert the `Account__r` column to a dictionary with the external id field that you are using to look up the Account object. This is because of the way relationships work in Salesforce. If you are somewhat familiar with how the platform's database works, you will notice that for the row that is going to be inserted the column that references the lookup object is transformed to an object, in a similar way that in Apex you use the relationship field to access the fields of the related object. This works for standard objects as well, where instead of *ContactId* you would use just *Contact*.
```

Comment:
Finding this sentence just a bit unclear. Unsure how to suggest an improvement but for someone unexperienced in Apex its hard to envision what being 'transformed to an object' means.

Reply:
I made some changes to this PR so that the spelling is fixed and this part is a little better explained.